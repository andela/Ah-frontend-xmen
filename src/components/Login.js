import React from 'react';

const Login = () => (
  <form style={{ padding: '43px' }}>
    <label htmlFor="username">
          Username:
      <input type="text" id="username" placeholder="Username" />
    </label>
    <br />
    <label htmlFor="password">
          Password:
      <input type="text" id="password" placeholder="Password" />
    </label>
    <br />
    <input type="submit" value="Login" />
  </form>
);

export default Login;
