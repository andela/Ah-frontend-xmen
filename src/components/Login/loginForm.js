/* eslint-disable jsx-a11y/anchor-is-valid */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Login = (props) => {
  const { onSubmit } = props;
  const { errors } = props;
  const { isSuccessful } = props;
  const emailError = errors ? errors.email : '';
  const passwordError = errors ? errors.password : '';
  const generalError = errors ? errors.error : '';
  const successMessage = isSuccessful ? 'Successfully logged in.' : '';

  return (
    <div className="card text-center mx-auto mt-4 border-0" id="loginCard">
      <div className="card-body text-center">
        <div className="row">
          <span className="col-md-5"><hr /></span>
          <span className="col-md-2 text-center text-muted">or</span>
          <span className="col-md-5"><hr /></span>
        </div>
        <form className="mt-0" onSubmit={onSubmit}>
          <div className="form-group mb-2">
            <div className="input-group input-group-alternative">
              <div className="input-group-prepend">
                <span className="input-group-text"><i className="fa fa-user" /></span>
              </div>
              <input name="email" className="form-control" placeholder="Email or Username" type="text" />
            </div>
          </div>
          <div>
            <p className="text-danger">{emailError}</p>
          </div>
          <div className="form-group">
            <div className="input-group input-group-alternative">
              <div className="input-group-prepend">
                <span className="input-group-text"><i className="fa fa-lock" /></span>
              </div>
              <input name="password" className="form-control" placeholder="Password" type="password" />
            </div>
          </div>
          <Link to="/password-reset"><p className="text-primary float-right">Forgot Password?</p></Link>
          <div>
            <p className="text-danger">{passwordError}</p>
          </div>
          <div>
            <button type="submit" className="btn btn-block button-primary my-4">Sign in</button>
          </div>
          <p>
            Don&#39;t have an account?&nbsp;
            <a href="#" onClick={props.onOpenSignup}>Sign up</a>
          </p>
          <div>
            <p className="text-danger">{generalError}</p>
            <p className="text-success">{successMessage}</p>
          </div>
        </form>
      </div>
    </div>
  );
};


Login.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  errors: PropTypes.shape({
    email: PropTypes.any,
    password: PropTypes.any,
    error: PropTypes.any,
  }),
};

Login.defaultProps = {
  errors: {},
};

export default Login;
