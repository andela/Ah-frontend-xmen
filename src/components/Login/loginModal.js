import React from 'react';
import Modal from 'react-responsive-modal';
import PropTypes from 'prop-types';
import LoginView from '../../views/loginView/LoginView';

const LoginModal = ({ open, onClose, onOpenSignup }) => (
  <div>
    <Modal open={open} onClose={onClose} classNames={{ modal: 'col col-md-6 col-sm-12 col-lg-4 col-offset-lg-3 col-offset-4' }}>
      <LoginView onOpenSignup={onOpenSignup} />
    </Modal>
  </div>
);

LoginModal.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default LoginModal;
