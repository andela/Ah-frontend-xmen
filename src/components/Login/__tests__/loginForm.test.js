import { shallow } from 'enzyme';
import React from 'react';
import Login from '../loginForm';


describe('login form on login success', () => {
  const props = {
    onSubmit: jest.fn(),
    errors: null,
    isSuccessful: true,
    emailError: '',
    passwordError: '',
    generalError: '',
    successMessage: 'Successfully logged in',
  };
  it('should render without crashing', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('login form on login fail', () => {
  const props = {
    onSubmit: jest.fn(),
    errors: {},
    isSuccessful: false,
    emailError: 'error',
    passwordError: 'error',
    generalError: 'error',
    successMessage: '',
  };
  it('should render without crashing', () => {
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
