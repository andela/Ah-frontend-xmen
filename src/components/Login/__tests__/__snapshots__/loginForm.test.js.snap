// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form on login fail should render without crashing 1`] = `ShallowWrapper {}`;

exports[`login form on login success should render without crashing 1`] = `ShallowWrapper {}`;
